### ISSUE TITLE
exp:channel:form removes config items set by ee()->config->set_item()

### ISSUE THREAD

**Issue by smaida** [ExpressionEngine/ExpressionEngine#621]
**Description of the problem**
We recently updated CartThrob to set config variables globally on the core_boot hook. The EE documentation suggests that we should be able to set config items with ee()->config->set_item() and then retrieve them later with ee()->config->item().

This seems to be true broadly except that there is at least one instance where EE replaces the config array without preserving any custom items set on the array. 

**How To Reproduce**
I've been able to reproduce this in CartThrob by using a simple template like so:
`{exp:channel:form channel="wish_list"}
{/exp:channel:form}

{exp:cartthrob:checkout_form return="store/order_complete" gateway="Cartthrob_dummy"}
{/exp:cartthrob:checkout_form}` 

That said, I think you can prove this without installing CartThrob. 

Modifying the switch_site method in Channel_form_lib confirms that items are removed.  Perhaps this is intended behavior. If it is, this is probably worth clarifying in the docs.

```
	private function switch_site($site_id)
	{
		ee()->config->set_item('site_id', $site_id);
                ee()->config->set_item('testing', 'peanuts');
                var_dump(ee()->config->item('testing'));
		ee()->config->site_prefs('', $site_id);
                var_dump(ee()->config->item('testing'));die();
	}
```

**Comment by ops-andy** [comment]
@intoeetive can you confirm this please.

**Comment by intoeetive** [comment]
@smaida am I correct assuming you are not making cross-site submissions to other MSM site?
If so, the following change should work - can you confirm that?
```	
	private function switch_site($site_id)
	{
		ee()->config->set_item('site_id', $site_id);
		ee()->config->get_cached_site_prefs($site_id);
	}
```
Though if you have CartThrob configured on one site and the form is placed on another one, you'll probably have the same issue. In that case, you'd need to add config item to each MSM site.

I could suggest using `ee('Config')->getFile()->set($key, $value, TRUE)` to write preference directly into config file, but I wouldn't as if the config is write-protected you'll get issues

**Comment by smaida** [comment]
@intoeetive 

I can confirm that we're not making cross-site submissions and that the suggested change does resolve the issue.

FWIW, I do plan on reworking how we're handling our own config data. Also, that is a helpful note regarding Channel_form_lib. We've been slowly migrating away from legacy EE can CartThrob code, so I'll note this in our backlog.

**Comment by intoeetive** [comment]
The fix has been released in EE 5.4.0 and 6.0.0
